stage0/layout.ld

```diff
@@ -213,9 +220,59 @@ SECTIONS {
      * https://github.com/qemu/qemu/blob/master/docs/specs/sev-guest-firmware.rst
      * and EDK2 source code:
      * https://github.com/tianocore/edk2/blob/master/OvmfPkg/ResetVector/Ia16/ResetVectorVtf0.asm
+     *
+     * Unfortunately it doesn't look like `SIZEOF()` works here, so we have to keep track of the data
+     * structure size by hand to ensure that it ends at the correct location in memory.
      */
-    .guid_tables TOP - 0x20 - 0x28 : {
+    .guid_tables TOP - 0x20 - (16 + 3*12 + 2*22 + 18) : {
+        /* SEV metadata. This data structure is not part of the GUIDed tables, but it makes sense to have
+         * them together at the end of the file, thus we put them in the .guid_tables section along with
+         * the GUIDed tables themselves. The data structure is defined in the EDK2 source:
+         * https://github.com/tianocore/edk2/blob/master/OvmfPkg/ResetVector/X64/OvmfSevMetadata.asm
+         *
+         * Size: header is 16 bytes + a number of 12-byte sections.
+         */
+        HIDDEN(sev_guided_structure_start = .);
+        /* Header: uint32 signature ("ASEV"), uint32 length, uint32 version, uint32 number of sections */
+        BYTE(0x41) BYTE(0x53) BYTE(0x45) BYTE(0x56)
+        LONG(sev_guided_structure_end - sev_guided_structure_start)
+        LONG(1)
+        LONG((sev_guided_structure_end - sev_guided_structure_start - 16) / 12)
+        /* Each section has the basic structure of uint32 address, uint32 length, uint32 type
+         * The locations of these pages have been chosen to be the same as EDK2:
+         * https://github.com/tianocore/edk2/blob/c05a218a9758225ddf94eedb365633f2154551da/OvmfPkg/OvmfPkgX64.fdf
+         * Thankfully none of these clashed with existing data structures that we're setting up.
+         */
+        /* Unmeasured page location */
```

Comment:
What will the unmeasured section used for? Will this be used for generating the boot parameter page in future?

Reply:
At this point it's unused. The main reason I added it was that our VMM refuses to boot if you don't have at least one unmeasured section, so there we are.

EDK2 sets up two or three unmeasured sections; I'll need to do some more digging to find out what they're used for in there.